cmd/buildkitd/main_windows.go

```diff
@@ -19,14 +21,18 @@ func listenFD(_ string, _ *tls.Config) (net.Listener, error) {
 	return nil, errors.New("listening server on fd not supported on windows")
 }
 
-func getLocalListener(listenerPath string) (net.Listener, error) {
-	pc := &winio.PipeConfig{
+func getLocalListener(listenerPath, secDescriptor string) (net.Listener, error) {
+	if secDescriptor == "" {
 		// Allow generic read and generic write access to authenticated users
 		// and system users. On Linux, this pipe seems to be given rw access to
 		// user, group and others (666).
 		// TODO(gabriel-samfira): should we restrict access to this pipe to just
 		// authenticated users? Or Administrators group?
-		SecurityDescriptor: "D:P(A;;GRGW;;;AU)(A;;GRGW;;;SY)",
+		secDescriptor = "D:P(A;;GRGW;;;AU)(A;;GRGW;;;SY)"
```

Comment:
The default can have just the system user allowed, and we can remove `Authenticated users`. This way we have a "secure default".

So this can be replaced with:

```go
secDescriptor = "D:P(A;;GRGW;;;SY)"
```

Or better yet, the SDDL you have in `groupToSecurityDescriptor` is also OK. It allows builtin administrators and system user `Generic All`.

Reply:
This will only be used in `traceController` listener. The buildkitd socket securitydescriptor will always be initialized to `groupToSecurityDescriptor("")` if not set.

Happy to change that but I don't really understand why this default is different than the one in `groupToSecurityDescriptor` (that I copied from moby/moby).

Reply:
The bit:

`(A;;GRGW;;;AU)`

means allow generic read and write for `Authenticated users`. That means all users except `guest`. The one in `groupToSecurityDescriptor` allows `generic all` to  the `builtin\administrators` group and the `LocalSystem` user, both of which are essentially administrators of the system, which is desirable in both cases (trace controller listener as well).

But if the trace controller is not always started, I guess it's fine to leave it as it is.